dist/screeps-tests.ts

```diff
@@ -1030,8 +1030,165 @@ function resources(o: GenericStore): ResourceConstant[] {
     EXTENSION_ENERGY_CAPACITY[Game.rooms.myRoom.controller!.level];
 
     REACTIONS[Object.keys(creep.carry)[0]];
+}
+
+{
+    // Test the BOOSTS constant
+
+    // Can be used with a body part, returns a record of mineral -> boosted property -> level
+    const c = BOOSTS[creep.body[0].type];
+
+    // Can be used with all body part types, returns undefined
+    const undef = BOOSTS["claim"];
+
+    // Can still be iterated over
+    for (const bodyPart of Object.keys(BOOSTS) as BodyPartConstant[]) {
+        const boosts = BOOSTS[bodyPart];
+        for (const mineral of Object.keys(boosts) as MineralBoostConstant[]) {
+            const upgrades = boosts[mineral];
```

Comment:
Without the fixes, this fails with
```
Element implicitly has an 'any' type because expression of type 'MineralBoostConstant' can't be used to index type '{ [boost: string]: { [action: string]: number; }; } | { UO: { harvest: 3; }; UHO2: { harvest: 5; }; XUHO2: { harvest: 7; }; LH: { build: 1.5; repair: 1.5; }; LH2O: { build: 1.8; repair: 1.8; }; XLH2O: { build: 2; repair: 2; }; ... 5 more ...; XGH2O: { ...; }; } | ... 5 more ... | { ...; }'.
  Property 'UH' does not exist on type '{ [boost: string]: { [action: string]: number; }; } | { UO: { harvest: 3; }; UHO2: { harvest: 5; }; XUHO2: { harvest: 7; }; LH: { build: 1.5; repair: 1.5; }; LH2O: { build: 1.8; repair: 1.8; }; XLH2O: { build: 2; repair: 2; }; ... 5 more ...; XGH2O: { ...; }; } | ... 5 more ... | { ...; }'.ts(7053)
```